kitsune/search/views.py

```diff
@@ -54,10 +67,79 @@ def jsonp_is_valid(func):
     return func_regex.match(func)
 
 
-# Search "Expires" header format
-EXPIRES_FMT = '%A, %d %B %Y %H:%M:%S GMT'
+def cache_control(resp, cache_period):
+    """Inserts cache/expires headers"""
+    resp['Cache-Control'] = 'max-age=%s' % (cache_period * 60)
+    resp['Expires'] = (
+        (datetime.utcnow() + timedelta(minutes=cache_period))
+        .strftime('%A, %d %B %Y %H:%M:%S GMT'))
+    return resp
+
+
+def _es_down_template(request, *args, **kwargs):
+    """Returns the appropriate "Elasticsearch is down!" template"""
+    return 'search/mobile/down.html' if request.MOBILE else 'search/down.html'
+
+
+def build_results_list(pages, is_json):
+    """Takes a paginated search and returns results List
+
+    Handles wiki documents, questions and contributor forum posts.
+
+    :arg pages: paginated S
+    :arg is_json: whether or not this is generated results for json output
+
+    :returns: list of dicts
+
+    """
+    offset = pages.start_index()
+    results = []
+    for i, doc in enumerate(pages):
```

Comment:
Since the only place you use `i` is to calculate the rank by adding it to `offset`, you could use `enumerate`'s second argument to get `rank` directly.

```python
>>> list(enumerate('abc'))
[(0, 'a'), (1, 'b'), (2, 'c')]
>>> list(enumerate('abc', 5))
[(5, 'a'), (6, 'b'), (7, 'c')]
```